src/huggingface_hub/hf_api.py

```diff
@@ -4772,6 +4772,92 @@ def delete_file(
             parent_commit=parent_commit,
         )
 
+    @validate_hf_hub_args
+    def delete_files_r(
+        self,
+        patterns: List[str],
+        repo_id: str,
```

Comment:
```suggestion
        repo_id: str,
        delete_patterns: List[str],
```

I'd place `repo_id` first (as in the command line) and rename to `delete_patterns` to be more explicit.